src/FakeItEasy/Core/TypeCatalogue.cs

```diff
@@ -91,7 +91,7 @@ private static IEnumerable<Assembly> GetAllAssemblies(IEnumerable<string> extraA
                     // Exclude the ReflectionOnly assemblies because we may fully load them later.
                     .Where(a => !a.ReflectionOnly)
 #endif
-                    .Where(a => !a.IsDynamic)
+                    .Where(a => !a.IsDynamic && !a.Name().Equals("DynamicProxyGenAssembly2", StringComparison.OrdinalIgnoreCase))

```

Comment:
I don't get it. From what I can tell, `DynamicProxyGenAssembly2` is dynamic, so I can't tell why this is needed.
It's not that I don't believe the problem exists. I'm just not sure why it does…